src/main/java/by/itechart/api/exception/UserResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,40 @@
+package by.itechart.api.exception;
+
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class UserResponseEntityExceptionHandler extends
+        ResponseEntityExceptionHandler {
+    @ResponseBody
+    @ExceptionHandler(UserNotFoundException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    protected ExceptionInfo handleUserNotFoundException(UserNotFoundException ex) {
+        String bodyOfResponse = ex.getMessage();
+        return new ExceptionInfo(bodyOfResponse);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException ex, HttpHeaders headers, HttpStatus status, WebRequest request) {
```

Comment:
Why do you return `Object` and not `ExceptionInfo` itself as in the handler above?

Reply:
The problem is when I create a custom method to handle `MethodArgumentNotValidException` that will return our `ExceptionInfo` there will be an error about the ambiguity of `ExceptionHandler` because `MethodArgumentNotValidException` is already defined with `@ExceptionHandler` annotation in the `ResponseEntityExceptionHandler` class. 
Here are the details: https://stackoverflow.com/questions/51991992/getting-ambiguous-exceptionhandler-method-mapped-for-methodargumentnotvalidexce